### DISCUSSION TITLE
Week 4 Check-in (short week)

### DISCUSSION THREAD

**Discussion by eeholmes** [UW-Upwelling-Project/Summer-2023#8] [General]
# Eli

* Set up https://github.com/Hackweek-ITCOocean for prepping our tutorials.
    - Add Minh and Jiarui and discuss how to create new tutorials
* Contacted OHW to get the participation of Minh (virtual) and Jiarui (in-person) fixed.

## Notes

* AGU 2020 tutorials https://github.com/pangeo-gallery/osm2020tutorial

**Comment by minhphan03** [comment]
# Wednesday's update:
* Downloaded, cleaned, and synthesized all variables

**Reply by eeholmes** [reply]
Try different chunking on Zarr files to see what has faster access.

https://discourse.pangeo.io/t/understanding-optimal-zarr-chunking-scheme-for-a-climatology/2335

https://discourse.pangeo.io/t/netcdf-to-zarr-best-practices/1119/13

**Reply by minhphan03** [reply]
https://discourse.pangeo.io/t/processing-large-too-large-for-memory-xarray-datasets-and-writing-to-netcdf/1724/3 appending by time, chunk of every 100 days.

**Comment by NaNa7Miiii** [comment]
### Jiarui Wed & Thurs Check-In:
Summarized all the data from **1979-01** to **2023-05** into .nc files in this format:
<img width="726" alt="Screenshot 2023-07-06 at 4 02 40 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/1a1f4b8b-1b5d-48a4-8ce4-06607e57953e">
The SST Diff is the difference between each pair (a point on the 100km buffer along the bounding box and its paired (nearest) point on the 300km buffer along the coastal line), we can apply a flexible threshold on those differences to get labels. Here is an example:
```
import xarray as xr
import pandas as pd

# Open the dataset and convert it to a pandas DataFrame
temp = xr.open_dataset('shared/sst_data/201407_sst_diff.nc')
temp_df = temp.to_dataframe()

# Convert the 'time' column to date only format
temp_df.index = pd.to_datetime(temp_df.index).date

# Reset index to create 'time' column
temp_df = temp_df.reset_index()

# Create a new column 'entry_id' which enumerates each entry within each date
temp_df['entry_id'] = temp_df.groupby('time').cumcount()

# Filter dataframe for rows where 'entry_id' is 22, 23 or 24
filtered_df = temp_df[temp_df['entry_id'].isin([22,23,24])]

# Create a label column - set to 1 if SST_Diff > threshold, else set to 0
threshold = 1
filtered_df['label'] = np.where(filtered_df['SST_Diff'] > threshold, 1, 0)

# Group by 'time' and check if all 'label' in a group is 1, if so, set group_label to 1, else 0
group_labels = filtered_df.groupby('time')['label'].apply(lambda x: 1 if x.sum() == 3 else 0)
```
<img width="219" alt="Screenshot 2023-07-06 at 4 08 01 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/7fab8515-47ca-484a-9a6e-77e31fec4730">

Visualization on heatmap (This example uses data from 1979-01)
<img width="910" alt="Screenshot 2023-07-06 at 4 05 30 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/8677dd46-1263-48a0-a8c5-6f865e728bc9">

**Comment by NaNa7Miiii** [comment]
Jiarui Thursday Checkin:
* Updated the threshold data, and saved the files into the `shared/sst_diff` folder
* Created a tutorial for a demo using LSTM to do training, saved it as `shared/jiarui-notebooks/Demo_LSTM.ipynb`
* Used all of the data from 2010-2020 to do training / testing for the LSTM model

**Comment by minhphan03** [comment]
Finished 1 tutorial (NASA Earthdata downloader)
Working on the next (ERDDAP downloader)
Added metadata and rechunk everything

**Comment by eeholmes** [comment]
# Friday

## Eli

* Met with Jim C at 2i2c re setting up a JupyterHub for the INCOIS hackweek.
* Sent some emails trying to get funding

## Minh

* Worked on ERDDAP tutorial

## Jiarui

* Worked on tutorial for image classification
* worked on the threshold

**Reply by eeholmes** [reply]
Look into dask-tensorflow

* https://ml.dask.org/keras.html
* https://matthewrocklin.com/blog/work/2017/02/11/dask-tensorflow
